### PR TITLE
WIP: Support non-alle repos

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,35 +47,75 @@ npm install @ianwremmel/clark
 
 ## Usage
 
-The key benefit of clark over [lerna](https://lernajs.io/) and similar tools comes from adhering to the directory layout described in [alle](https://github.com/boennemann/alle). In other words, all of your package **must** be kept in `./packages/node_modules` and the name each each `package.json` **should** match the subfolder path (yes, this includes the org/user scope if present).
+Unlike [lerna](https://lernajs.io/) or similar tools, `clark` let's you keep track of your dependencies all in your main `package.json` (key benefits being significantly faster `npm install` times and the ability to use greenkeeper). In order to get this benefit, however, you'll need to follow one of two patterns ([alle](#alle) or [non-alle](#non-alle), described below). Once you pick one of those patterns and configure your repository accordingly, you can use `clark hoist`, to move your deps from you individual subpackages to your repo root.
```
Comment (kourge) [inline README.md]: type: lets

File: README.md [hunk 2]
```diff
@@ -43,35 +47,75 @@ npm install @ianwremmel/clark
...
 
-### List all commands
+### Alle
+
+[Alle](https://github.com/boennemann/alle) was originally described as just an example of how things _could_ work, before eventually being enacted by [pouchdb](). In order to follow the Alle pattern, all of your package **must** be kept in `./packages/node_modules` and the name of each `package.json` **should** match the subfolder path (yes, this includes the org/user scope if present).
+
+**Benefits**
+
+*   Alle is symlink free. It relies on the behavior of `require()` inside a `node_modules` directory to search both up the tree and in sibling folders, thus letting your packages find each automatically.
```
Comment (kourge) [inline README.md]: reword: "find each automatically"

File: README.md [hunk 3]
```diff
@@ -43,35 +47,75 @@ npm install @ianwremmel/clark
...
+
+**Caveats**
+
+*   Many tools have hardcoded excludes for `node_modules` and some can't be overridden at all. Perhaps most problematically, GitHub PR will collapse most of your diffs assuming that anything in `node_modules` is vendored. GitHub language stats also get confused.
+*   If you already have an established project, moving every folder can be problematic.
+
+### Non-Alle
+
+When npm encounters a package version that's simply a file path (e.g. `"my-package": "file:./packages/my-package"`), it will symlink it into `./node_modules`. By putting all of our local node_modules in the top-level `package.json`, we can expose our local packages to eachother withought making any other repo changes.
```
Comment (kourge) [inline README.md]: * typo: each other
* typo: without

File: src/lib/util.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+interface AnyObject {
+  [key: string]: any;
+}
+
+/**
+ * Sorts an object
+ * @param obj
+ */
+export function sortObject(obj: object): object {
```
Comment (kourge) [inline src/lib/util.ts]: What is the semantics of this function meant to be?

File: src/lib/project.ts [hunk 5]
```diff
@@ -41,6 +41,20 @@ export async function hasRc(rootDir: string): Promise<boolean> {
...
 }
 
+/**
+ * Iterates overall package paths and, depending on the presence or absence of
+ * "packages/node_modules", determines if this is a alle or non-alle monorepo.
+ * @param paths
+ */
+export function isAlleRepo(paths: string[]): boolean {
+  for (const packagePath of paths) {
+    if (!packagePath.includes('packages/node_modules')) {
```
Comment (kourge) [inline src/lib/project.ts]: Use `Array.prototype.every`

File: src/lib/project.ts [hunk 6]
```diff
@@ -41,6 +41,20 @@ export async function hasRc(rootDir: string): Promise<boolean> {
   return contains;
 }
 
+/**
+ * Iterates overall package paths and, depending on the presence or absence of
```
Comment (kourge) [inline src/lib/project.ts]: Document _what_ over _how_. "Determines whether the given paths describe an alle monorepo by checking for a `packages/node_modules` path in all of them."

File: src/lib/util.ts (hunk 4 above)
Comment (ianwremmel) [reply src/lib/util.ts]: What do you mean? Does changing the description to `Returns a new clone of 'obj' with its keys sorted alphabetically` answer that question?

**Pull request merged**